src/plugin/file/kml/ui/kmlnode.js

```diff
@@ -304,6 +304,14 @@ plugin.file.kml.ui.KMLNode.prototype.onFeatureChange = function(event) {
         this.dispatchEvent(new os.events.PropertyChangeEvent('icons'));
         this.dispatchEvent(new os.events.PropertyChangeEvent(os.annotation.EventType.CHANGE));
         break;
+      case 'colors':
+        var source = /** @type {plugin.file.kml.KMLSource} */ (os.feature.getSource(this.feature_));
+        var rootNode = source.getRootNode();
+        var children = rootNode.getChildren();
+        for (var i = 0; i < children.length; i++) { // update icon color
+          children[i].dispatchEvent(new os.events.PropertyChangeEvent('icons'));
```

Comment:
This is still dispatching change events on all of the children of the root, which seems incorrect. The node that receives the color change event (which is this node, as it is the one tied to the feature) should be the only one that needs to dispatch a `new os.events.PropertyChangeEvent('icons')`.

Reply:
So something more like this? `this.dispatchEvent(new os.events.PropertyChangeEvent('icons'));`